sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -316,16 +316,39 @@ static const struct snd_kcontrol_new controls[] = {
 	SOC_DAPM_PIN_SWITCH("Speaker"),
 };
 
+static int set_amp_tdm_slots(struct snd_soc_pcm_runtime *rtd, bool tdm_mode)
+{
+	struct snd_soc_dai *codec_dai;
+	int ret = 0;
+	int val;
+	int i;
+
+	for_each_rtd_codec_dai(rtd, i, codec_dai) {
+		if (tdm_mode)
+			/* tx_mask 0, rx_mask with only one bit, single slot) */
+			val = snd_soc_dai_set_tdm_slot(codec_dai, 0, 1 << i, 1, 0);
+		else
+			/* tx_mask 0, rx_mask 0x3, 2 slots */
+			val = snd_soc_dai_set_tdm_slot(codec_dai, 0, 0x3, 2, 0);
+		if (!ret)
+			ret = val;
```

Comment:
Should we return error right away and exit the loop?

Reply:
yes, not sure what I was thinking here.